tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -226,55 +229,121 @@ def _check_ssm_success(ssm_client, command_id, instance_id):
     ).is_true()
 
 
+class DirectoryStackSharedFixture(SharedFixture):
```

Comment:
[BLOCKING] I think the easiest way to use SharedFixture is by using the decorated `@xdist_session_fixture`

See example: https://github.com/aws/aws-parallelcluster/blob/05eee1539124d84e3a55aeb9e90e83aa7bed337b/tests/integration-tests/conftest_networking.py#L285

Why can't we reuse the same approach we are currently using for shared VPC stacks?